python/tank/pipelineconfig.py

```diff
@@ -83,59 +81,118 @@ def __init__(self, pipeline_configuration_path):
 
-        # cache fields lazily populated on getter access
-        self._clear_cached_settings()
+        if pipeline_config_metadata.get("bundle_cache_fallback_roots"):
+            self._bundle_cache_fallback_paths = pipeline_config_metadata.get("bundle_cache_fallback_roots")
+        else:
+            self._bundle_cache_fallback_paths = []
 
         # Populate the global yaml_cache if we find a pickled cache
         # on disk.
-        # TODO: Discuss and possibly implement a mechanism to copy
-        # the global pickled cache stored in the config down to
-        # local storage to speed up reading in subsequent sessions.
+        # TODO: For immutable configs, move this into bootstrap
         self._populate_yaml_cache()
-        
+
+        # run init hook
         self.execute_core_hook_internal(constants.PIPELINE_CONFIGURATION_INIT_HOOK_NAME, parent=self)
 
     def __repr__(self):
         return "<Sgtk Configuration %s>" % self._pc_root
 
-    def _clear_cached_settings(self):
+    ########################################################################################
+    # handling pc metadata
+    
+    def _get_metadata(self):
         """
-        Force the pc object to reread its settings from disk.
-        Call this if you have made changes to config files and 
-        want these to be picked up. The next time settings are needed,
-        these will be automatically re-read from disk.
+        Loads the pipeline config metadata (the pipeline_configuration.yml) file from disk.
+        
+        :param pipeline_config_path: path to a pipeline configuration root folder
+        :returns: deserialized content of the file in the form of a dict.
         """
-        self._project_id = self._UNDEFINED_PROJECT_ID
-        self._pc_id = None
-        self._pc_name = None
-        self._published_file_entity_type = None
-        self._cache_folder = None
-        self._path_cache_path = None
-        self._use_shotgun_path_cache = None
-
-    def _load_metadata_from_sg(self):
+    
+        # now read in the pipeline_configuration.yml file
+        cfg_yml = os.path.join(self.get_config_location(), 
+                               "core", 
+                               "pipeline_configuration.yml")
+    
+        if not os.path.exists(cfg_yml):
+            raise TankError("Configuration metadata file '%s' missing! "
+                            "Please contact support." % cfg_yml)
+    
+        fh = open(cfg_yml, "rt")
+        try:
+            data = yaml.load(fh)
+            if data is None:
+                raise Exception("File contains no data!")
+        except Exception, e:
+            raise TankError("Looks like a config file is corrupt. Please contact "
+                            "support! File: '%s' Error: %s" % (cfg_yml, e))
+        finally:
+            fh.close()
+    
+        return data
+    
+    def _update_pipeline_config(self, updates):
```

Comment:
minor: Maybe this should be called `_update_metadata()` for consistency with `_get_metadata()`.